### PR TITLE
DAOS-12651 test: add nlt test for dfs container check

### PR DESCRIPTION
**Pull request by mchaarawi**
Skip-func-test: true
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by mchaarawi** [comment]
oops.. shouldn't have asked for review yet.
seems like this is failing in the new NLT test added. although it passed locally. will investigate

File: src/tests/SConscript [hunk 1]
```diff
@@ -108,7 +108,9 @@ def scons():
     denv.d_program('simple_array', 'simple_array.c', LIBS=libs_client)
     denv.d_program('simple_obj', 'simple_obj.c', LIBS=libs_client)
     denv.d_program('simple_dfs', 'simple_dfs.c', LIBS=libs_client)
-    denv.d_program('daos_mw_fi', 'daos_mw_fi.c', LIBS=libs_client)
+    libs_client.remove('daos_tests')
```
Comment (ashleypittman) [inline src/tests/SConscript:111]: I'm not aware of anywhere else in the build where we use .remove() however this is only on a list so should be OK.  The alternative would be moving lines 112-113 to line 107 and then calling libs_client.add('daos_tests') before building the others which I think would be cleaner.

File: utils/node_local_test.py [hunk 2]
```diff
@@ -3014,6 +3014,148 @@ def test_cont_clone(self):
...
+        print(cmd)
+        rc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            env=cmd_env, check=False, cwd=self.dfuse.dir)
+        if rc.stderr != b'':
+            print('Stderr from command')
+            print(rc.stderr.decode('utf-8').strip())
+        if rc.stdout != b'':
+            print(rc.stdout.decode('utf-8').strip())
+
+        cmd = ['fs', 'check', pool, cont, '--flags', 'print,relink', '--dir-name', 'lf']
```
Comment (ashleypittman) [inline utils/node_local_test.py:3130]: Is the output here deterministic, could you check for a known-string or does it contain object IDs?

File: utils/node_local_test.py [hunk 3]
```diff
@@ -3014,6 +3014,148 @@ def test_cont_clone(self):
...
+        rc = run_daos_cmd(self.conf, cmd)
+        print(rc)
+
```
Comment (ashleypittman) [inline utils/node_local_test.py:3133]: You could probably call the command here again and check/verify that it doesn't find anything a second time, the output there should be deterministic so you could verify it.

**Review by ashleypittman — APPROVED** [review]

File: utils/node_local_test.py (hunk 3 above)
Comment (mchaarawi) [reply utils/node_local_test.py:3133]: yes, but trying to get it to pass in CI first

File: utils/node_local_test.py (hunk 2 above)
Comment (mchaarawi) [reply utils/node_local_test.py:3133]: it has both. the last line should be deterministic so i can check that.

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**